addons/website_slides/models/slide_slide.py

```diff
@@ -877,16 +897,48 @@ def _action_set_completed(self, target_partner):
             'vote': 0,
             'completed': True} for new_slide in new_slides])
 
+        self.env['slide.channel'].invalidate_cache(['completed', 'completion'], ids=self.channel_id.ids)
+
+        return True
+
+    def action_set_uncompleted(self):
+        if any(not slide.channel_id.is_member for slide in self):
+            raise UserError(_('You cannot mark a slide as uncompleted if you are not among its members.'))
+
+        if any(not slide.can_be_undone for slide in self):
+            raise UserError(_('Those slides can not be undone.'))
```

Comment:
Fits my point earlier, this line can never be reached since "can_be_undone" only relies on membership, so it will fail here above first.

Maybe you wanted to be flexible and allow overrides to specify more condition for "can_be_undone", but it does not seem necessary and makes the code less readable in my opinion.